test/serializer_test.dart

```diff
@@ -248,26 +247,23 @@ main() {
 
     test("simple test - fromMap", () {
       ModelA a =
-      Serializer.fromMap({"@dart_type": "ModelA", "foo": "toto"});
+      Serializer.fromMap({"foo": "toto"}, ModelA);
 
       expect(ModelA, a.runtimeType);
       expect("toto", a.foo);
     });
 
-
-
     test("Map fromMap Map", () {
```

Comment:
Without type_info_key, an anonymous map should embed the same type.

To support the following notation, you should add a class `Titi`
```dart
{"titi": new ModelA().toMap(), "foo": "bar" };

class Titi {
  ModelA titi;
  String foo;
}
```

If type_info_key is kept as an optional parameter, the solution would be reassessed.